tests/test_loaders.py

```diff
@@ -244,3 +255,66 @@ def test_ansible_lint_exception(mocked_popen, loader_role):
     mocked_popen.return_value.wait.return_value = 1
     res = list(loader_role._lint_role('.'))
     assert 'Exception running ansible-lint' in res[0]
+
+
+def test_get_metadata_file(temp_root, loader_role):
+    res = loader_role._get_metadata_file(temp_root)
+    assert res is None
+
+    meta_dir = os.path.join(temp_root, 'meta')
+    os.mkdir(meta_dir)
+    with open(os.path.join(meta_dir, 'main.yml'), 'w'):
+        pass
+    res = loader_role._get_metadata_file(temp_root)
+    assert res == os.path.join(temp_root, 'meta', 'main.yml')
+
+
+def test_get_role_metadata_desc(temp_root, loader_role):
+    loader_role.rel_path = temp_root
+    meta_dir = os.path.join(temp_root, 'meta')
+    os.mkdir(meta_dir)
+
+    res = loader_role._get_metadata_description()
+    assert res is None
+
+    with open(os.path.join(meta_dir, 'main.yml'), 'w') as fp:
+        fp.write(ROLE_METADATA)
+    res = loader_role._get_metadata_description()
+    assert res == 'Test description inside metadata'
+
+    with open(os.path.join(meta_dir, 'main.yml'), 'w') as fp:
+        fp.write(ROLE_METADATA_MISSING_DESC)
+    res = loader_role._get_metadata_description()
+    assert res is None
+
+
+def test_get_role_readme(temp_root, loader_role):
+    loader_role.root = temp_root
+    loader_role.rel_path = ''
+
+    with pytest.raises(exc.ContentLoadError):
+        loader_role._get_readme()
```

Comment:
I might split these tests into two tests, one for the null/empty case, and one for the case where it is expected to work.

ie
``` python
def test_role_loader_content_role_error_on_no_readme(temp_root, loader_role):
￼    loader_role.root = temp_root
￼    loader_role.rel_path = ''
￼
￼    with pytest.raises(exc.ContentLoadError):
￼        loader_role._get_readme()

def test_role_loader_get_readme(temp_root, loader_role):
    loader_role.root = temp_root
    loader_role.rel_path = ''
    with open(os.path.join(temp_root, 'README.md'), 'w') as fp:
￼        fp.write('This is the role readme text')
￼   readme = loader_role._get_readme()
￼   assert readme.name == 'README.md'
￼   assert readme.text == 'This is the role readme text'
```